aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -131,14 +124,34 @@ Tensor true_divide(const Tensor& self, const Tensor& divisor) {
   return iter.output();
 }
 
-Tensor floor_divide(const Tensor& input, const Tensor& other) {
-  Tensor out = input / other;
-  return truncate(out);
+Tensor& floor_divide_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  auto iter = TensorIterator::binary_op(result, self, other,
+    /*check_mem_overlap=*/true);
+  div_stub(iter.device_type(), iter);
+
+  if (result.is_floating_point()) {
+    result.trunc_();
+  }
+
+  return result;
+}
+
+Tensor floor_divide(const Tensor& self, const Tensor& other) {
+  Tensor result;
+  auto iter = TensorIterator::binary_op(result, self, other,
+    /*check_mem_overlap=*/true);
+  div_stub(iter.device_type(), iter);
+
+  auto out = iter.output();
+  if (out.is_floating_point()) {
+    return out.trunc();
```

Comment:
nit: since this is out-of-place and we internally allocated the output, can't we do `trunc_()`?  I guess it depends on what we do for autograd -- what do we do?

Reply:
Yes we can! Great catch!

floor_divide doesn't have an autograd formula and this PR doesn't add one. I don't think it should, either, since we're already sneaking in a few additions (like sparse functionality) beyond the BC-breaking changes. The autograd formula for floor_divide would (based on our current formulas) return an all zeros grad tensor, too, because that's what torch.trunc() does.

Reply:
the part I was concerned about what the inplace change triggering a version counter problem (i.e. that the version counter gets incremented so we can't backprop through `floor_divide`, even to get the tensor of all zeros.  This would definitely not be a problem if we had a formula for floor_divide (that just set all zeros) because the output would be totally internal, but I'm unsure if it's a problem if we just rely on the autograd formulas for the underlying implementations.

Did you check if you can autograd through this version of floor_divide (to get all zeros?).

Reply:
I also realize I contradicted myself a bit in saying you should minimize the changes here and then asking why there's no sparse implementation :).

Reply:
You cannot backprop through floor_divide, you'll get an error when trying:

`RuntimeError: derivative for floor_divide is not implemented`

We could make it return all zeros in this PR (analogous to trunc). Would that address your concerns? It's a simple change. 

(No worries.)